src/main/java/wolox/training/models/User.java

```diff
@@ -193,6 +193,31 @@ public void removeBookToUserById(Long id) {
      * @return Boolean true if the book is already in the library collection, if the book isn't in the library return false
      */
     private Boolean validateBook(Book book) {
-        return this.library.contains(book);
+        Book response = getBookFromCollection(book);
+        return this.library.contains(response);
+    }
+
+    /**
+     * This method is used to get the same reference to book by author in the collection library
+     * @param book: Book model class with the following attributes: (Book)
+     *     id: Id of the book (Long),
+     *     genre: Genre of the book (String),
+     *     author: Author of the book (String),
+     *     image: Image of the book (String),
+     *     title: Title of the book (String),
+     *     subtitle: Subtitle of the book (String),
+     *     publisher: Publisher of the book (String),
+     *     year: Year of the book (String),
+     *     pages: Pages of the book (Integer),
+     *     isbn: Isbn of the book (String).
+     * @return a {@link Book} with the same reference that the book in collection library if there is in there.
+     */
+    private Book getBookFromCollection(Book book) {
+        for (Book b: this.library) {
```

Comment:
The library is a list, maybe you can try the `.contains()` method instead of using a for 😃

Reply:
i tried it, but the book passed in the parameters have different reference to the book in the collection, althouth they had the same data and for that reason i couldn't verified successfully if the book was already in the library or remove from it :crying_cat_face: